sbi/inference/trainers/base.py

```diff
@@ -1059,3 +1070,30 @@ def check_if_proposal_has_default_x(proposal: Any):
             "x_o for training. Set it with "
             "`posterior.set_default_x(x_o)`."
         )
+
+
+ConditionalEstimatorType = TypeVar(
+    'ConditionalEstimatorType',
+    bound=ConditionalEstimator,
+    covariant=True,
+)
+
+
+class DensityEstimatorBuilder(Protocol[ConditionalEstimatorType]):
```

Comment:
I think these two should live in the `sbi/neural_nets/estimators/` module, e.g., in `base.py`. They are kind of orthogonal to the `trainers`